docs/content/SelectMenu.md

```diff
@@ -348,3 +348,34 @@ SelectMenu.Header components get `COMMON` system props. Read our [System Props](
 
 ### Component Props
 SelectMenu.Header components do not get any additional props besides system props.
+
+## SelectMenu.LoadingAnimation
+Use a `SelectMenu.LoadingAnimation` to add a loading animation inside of the SelectMenu.
+
+**Note**:  You will need to handle showing/hiding the appropriate modal content for your application during the loading state. We recommend always showing the `SelectMenu.Filter` and `SelectMenu.Header`  (if used) and hiding the rest of the modal content during the loading state.
+
+```jsx live
+<SelectMenu>
+  <Button as="summary">Projects</Button>
+  <SelectMenu.Modal>
+    <SelectMenu.Header>Projects</SelectMenu.Header>
+    <SelectMenu.Filter placeholder="Filter projects" value="" aria-label="Filter Projects"/>
+    {true ? <SelectMenu.LoadingAnimation/> :
```

Comment:
@BinaryMuse is there a way to set a variable in the code example here like `const loading = true` ? I tried adding it above line 358 but still got a `loading is undefined` error 🤔 I thought there was a way to do this but maybe not. If there is, I'd like to take everything from line 364 to line 370 and put it in a const so I can do this:

`{loading ? <SelectMenu.LoadingAnimation/> : modalContent}`
to make it look a little cleaner

Reply:
Unfortunately not; without [the `noinline` annotation](https://primer.style/doctocat/usage/live-code#no-inline) Doctocat treats the whole code block as a single expression.

Reply:
(But you could rewrite the example using `noinline`)